src/plugins/Tus.js

```diff
@@ -289,6 +289,10 @@ module.exports = class Tus extends Plugin {
 
     socket.on('progress', (progressData) => emitSocketProgress(this, progressData, file))
 
+    socket.on('error', (errData) => {
+      this.core.emitter.emit('core:upload-error', file.id, new Error(errData.error))
```

Comment:
nitpick, but its better if we are consistent: `this.core.emitter.emit` --> `this.core.emit`.